File: adventofcode/2024/01_2.cpp
#include <bits/stdc++.h>
#define int long long

using namespace std;

signed main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    #ifndef ONLINE_JUDGE 
        freopen("../input.txt", "r", stdin); 
        freopen("../output.txt", "w", stdout); 
    #endif
    const int LINES=1000;

    vector<int> x(LINES), y(LINES);
    map<int, int> rmp;
    for(int i=0; i<LINES; i++) {
        cin >> x[i] >> y[i];
        rmp[y[i]]++;
    }
    int ans = 0;
    for(int i=0; i<LINES; i++) {
        ans += x[i]*rmp[x[i]];
    }
    cout << ans;
    return 0;
}

File: adventofcode/2024/python/01_2.py
import sys
from collections import defaultdict

def main():
    sys.stdin = open("../input.txt", "r")
    sys.stdout = open("../output.txt", "w")
    
    LINES = 1000
    
    # Read input
    x = []
    y = []
    rmp = defaultdict(int)

    for _ in range(LINES):
        a, b = map(int, input().split())
        x.append(a)
        y.append(b)
        rmp[b]+=1
    
    # Sort the lists
    x.sort()
    y.sort()
    
    # Calculate the answer
    ans = 0
    for i in range(LINES):
        ans += x[i] * rmp[x[i]]
    
    print(ans)

if __name__ == "__main__":
    main()